### PR TITLE
Exclude script filter

### PR DESCRIPTION
**Pull request by ciddan**
Another proposed solution for #3. Uses filters in config to exclude scripts from rewrite.

### CODE REVIEW

**Comment by Menci** [comment]
Could you add a script like `/node_modules/jquery` to testcase?

**Comment by ciddan** [comment]
> Could you add a script like `/node_modules/jquery` to testcase?

Done!

**Comment by Menci** [comment]
I checked the code. It seems that Vite doesn't support loading `<script>`s from `/node_modules` directly. Let's revert it.

File: src/index.ts [hunk 1]
```diff
@@ -77,6 +83,9 @@ export function publicPath(options: Options): Plugin {
     apply: "build",
     configResolved(resolvedConfig) {
       viteConfig = resolvedConfig;
+      viteConfig.excludeScriptsFilter = options.excludeScripts
```
Comment (Menci) [inline src/index.ts]: I think we should append properties to `viteConfig`. Consider use `createFilter` in `processHtml` instead?

File: src/processing/html.ts [hunk 2]
```diff
@@ -4,6 +4,10 @@ import serialize from "serialize-javascript";
 import { ViteConfig, Options } from "..";
 
 export function processHtml(config: ViteConfig, options: Options, _fileName: string, html: string) {
+  if (!config.excludeScriptsFilter) {
```
Comment (Menci) [inline src/processing/html.ts]: Just `createFilter` from options here.

**Review by Menci — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (ciddan) [reply src/index.ts]: Done.

File: src/processing/html.ts (hunk 2 above)
Comment (ciddan) [reply src/processing/html.ts]: Done.

**Pull request merged**